runtime/oti/j9nonbuilder.h

```diff
@@ -5157,9 +5157,11 @@ typedef struct J9InternalVMFunctions {
 	BOOLEAN (*isCRIUSupportEnabled)(struct J9VMThread *currentThread);
 	BOOLEAN (*enableCRIUSecProvider)(struct J9VMThread *currentThread);
 	BOOLEAN (*isCheckpointAllowed)(struct J9VMThread *currentThread);
+	BOOLEAN (*isCheckpointAllowed_VM)(struct J9JavaVM *vm);
```

Comment:
Why do we need or want both `isCheckpointAllowed` and `isCheckpointAllowed_VM`?
Same question for `isDebugOnRestoreEnabled` / `isDebugOnRestoreEnabled_VM`.

Reply:
If you think the switch to the `_VM` signatures should be done separately, I suggest it should be done first (to avoid the unnecessary duplication).

Reply:
Opened
* https://github.com/eclipse-openj9/openj9/pull/20205

@dsouzai suggested holding it until https://github.com/eclipse-openj9/openj9/pull/20047 is merged.

This PR is a follow-up of https://github.com/eclipse-openj9/openj9/pull/19754, and is required for 0.48.

Existing code has a similar code pattern 
https://github.com/eclipse-openj9/openj9/blob/c8520fe02bc03346c5f2b37dc0f4c6c47392da33/runtime/vm/CRIUHelpers.cpp#L128-L138

Can this PR be merged first before the refactoring PR https://github.com/eclipse-openj9/openj9/pull/20205?